LibreHardwareMonitorLib/Hardware/Controller/Nzxt/KrakenV3.cs

```diff
@@ -1,218 +1,228 @@
-        }
-        else if (control.ControlMode == ControlMode.Default)
-        {
-            // There isn't a "default" mode with this fan, but a safe setting is 40%
-            FillTargetArray(_setFanTarget, 40);
-            _stream.Write(_setFanTarget);
-        }
-    }
-
-    public override void Close()
-    {
-        base.Close();
-        _stream?.Close();
-    }
-
-    private void ContinuousRead(object state)
-    {
-        byte[] buffer = new byte[_rawData.Length];
-        while (_stream.CanRead)
-        {
-            try
-            {
-                _stream.Write(_status_req); // Request status
-                _stream.Read(buffer); // This is a blocking call, will wait for bytes to become available
-
-                lock (_rawData)
-                {
-                    Array.Copy(buffer, _rawData, buffer.Length);
-                }
-            }
-            catch (TimeoutException)
-            {
-                // Don't care, just make sure the stream is still open
-                Thread.Sleep(500);
-            }
-            catch (ObjectDisposedException)
-            {
-                // Could be unplugged, or the app is stopping...
-                return;
-            }
-        }
-    }
-
-    public override void Update()
-    {
-        lock (_rawData)
-        {
-            if (_rawData[0] == 0x75 /*status response*/ && _rawData.Length >= 30)
-            {
-                _temperature.Value = _rawData[15] + (_rawData[16] / 10.0f);
-                _pumpRpm.Value = (_rawData[18] << 8) | _rawData[17];
-                _fanRpm.Value = (_rawData[24] << 8) | _rawData[23];
-
-                // The following logic makes sure the pump is set to the controlling value. This pump sometimes sets itself to 0% when instructed to a value.
-                if (!_controlling)
-                {
-                    _pump.Value = _rawData[19];
-                    _fan.Value = _rawData[25];
-                }
-                else if (_pump.Value != _rawData[19] || _fan.Value != _rawData[25])
-                {
-                    if (_pump.Value != _rawData[19])
-                    {
-                        float value = _pump.Value.GetValueOrDefault();
-                        FillTargetArray(_setPumpTarget, (byte)value);
-                        _stream.Write(_setPumpTarget);
-                    }
-
-                    if (_fan.Value != _rawData[25])
-                    {
-                        float value = _fan.Value.GetValueOrDefault();
-                        FillTargetArray(_setFanTarget, (byte)value);
-                        _stream.Write(_setFanTarget);
-                    }
-                }
-                else
-                {
-                    _controlling = false;
-                }
-            }
-        }
-    }
-}
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using HidSharp;
+
+namespace LibreHardwareMonitor.Hardware.Controller.Nzxt;
```

Comment:
Thanks for this PR!
Can you please reformat this document to address the odd whitespace here and there? Thanks! 
Is the rest now all done? I don't have these devices to test.

Reply:
code reformatted
I've successfully tested this on kraken X73 and on kraken elite.

Reply:
Thanks, nice PR 👍